### ISSUE TITLE
#line in the IDE - #nowarn, FindAllReferences and other things don't work

### ISSUE THREAD

**Issue by abelbraaksma** [dotnet/fsharp#9795]
#line causes many problems in the IDE

### `#nowarn` doesn't work in the presence of `#line`

```
#nowarn "25"
#line 1 "foo.fs"
let f 1 = 2
```

Expect: no warning

Actual: 
```
foo.fs(1,7): warning FS0025: Incomplete pattern matches on this expression. For example, the value '0' may indicate a case not covered by the pattern(s).
```

### FindAllReferences doesn't work

Take this code (presumably generated):
```fsharp
module File1

#nowarn "25"
#line 1 "foo.fs"
let Thing 1 = 2 
```

FindAllReferences on "Thing" 

Expected: one usage at the correct location in the given generated file (this is what C# does)

Actual: one usage on line 1 of File1.fs

**Comment by dsyme** [comment]
I guess this is really a request to have `#nowarn` work even when `'line` is used.  I'll rename

**Comment by dsyme** [comment]
I rejigged this issue to be a general issue about the problems `#line` causes in the IDE.

I started looking at this, and realised the problems are quite deep and may require significant rejig of how `#line` is implemented.

Following C#'s treatment, the spec we want appears to be

* The compiler reports diagnostics with `#line` applied
* When editing, diagnostics from document analysis have `#line` applied in their report in the error list, but double clicking on the error goes to the generated source code, not the `#line` source code. These diagnostics do not necessarily unify with diagnostics from the build.
* For all other IDE features, `#line` is ignored.  

Indeed we could probably just completely ignore `#line` in the IDE
